File: Backend/Events/Converters/Parties/AttributesConverter.cs
using Backend.Domain.Models.Parties.Digimons;
using Backend.Events.DTO.Party.Digimon;

namespace Backend.Events.Converters.Parties;

public static class AttributesConverter
{
    public static AttributesDTO ToDTO(Attributes attributes) => new()
    {
        Strength = attributes.Strength,
        Defense = attributes.Defense,
        Spirit = attributes.Spirit,
        Wisdom = attributes.Wisdom,
        Speed = attributes.Speed,
        Charisma = attributes.Charisma
    };
}

File: Frontend/src/events/converters/parties/digimons/attributes.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { AttributesDTO } from "@/events/dto/parties/digimons/attributes.dto";
import type { Attributes } from "@/models";

export class AttributesConverter {
  public static convert(attributesDto: DeepRequired<AttributesDTO>): Attributes {
    return {
      strength: attributesDto.strength,
      defense: attributesDto.defense,
      spirit: attributesDto.spirit,
      wisdom: attributesDto.wisdom,
      speed: attributesDto.speed,
      charisma: attributesDto.charisma,
    };
  }
}
